boot/bootstate20.go

```diff
@@ -73,6 +73,12 @@ type bootloaderKernelState20 interface {
 	// markSuccessfulKernel marks the specified kernel as having booted
 	// successfully, whether that kernel is the current kernel or the try-kernel
 	markSuccessfulKernel(sn snap.PlaceInfo) error
+	// setKernel sets the specified kernel as the next boot kernel, without
+	// the "try" logic. This shall be used only when we have already booted
+	// to a new kernel but for some reason we need to revert to the previous
+	// kernel (for instance, in a transactional update when the failing snap
+	// is not the kernel).
+	setKernel(sn snap.PlaceInfo) error
```

Comment:
Maybe `setNextKernelNocheck()`. Though I think we should try to use the same mechanism everywhere (that is, decide whether we want to add `isUndo` or some options to the existing methods, or to split out a method which does the operation without trying).

Reply:
Indeed, whether the code is shared it's an implementation detail, and if there's not much to share it might be better to keep the functions separate. Thanks for the name change!